packages/ses/src/global-object.js

```diff
@@ -14,7 +14,12 @@ export function initGlobalObject(
   globalObject,
   intrinsics,
   newGlobalPropertyNames,
-  { globalTransforms, nativeBrander, makeCompartmentConstructor },
+  {
+    globalTransforms,
+    nativeBrander,
+    makeCompartmentConstructor,
+    compartmentPrototype,
+  },
```

Comment:
Actually, I no longer understand why these may be `undefined`. Are we still trying to support a `lockdown` with no `Compartment` at all? This is a reasonable thing to do. I just don't know if we still are trying to do that, or if there is some other reason this might be `undefined`.

Reply:
Pushing a sequence of refactors that address this feedback. `makeCompartmentConstructor` has been promoted out of the options bag. `compartmentPrototype` has been rethreaded externally.